### ISSUE TITLE
Does a change of background pass Focus Appearance (Minimum)

### ISSUE THREAD

**Issue by lchandelier** [w3c/wcag#1278]
Hi,

Regarding this new criterion, I'm not sure the following implementation remains valid or not?

A black button on white background, becoming grey with a box-shadow around it on focus.
The grey color is `#404040` and the shadow is `box-shadow: #000 0 0 1px 1px;`

Thanks in advance for your help

Initial button
![initial-state](https://user-images.githubusercontent.com/13316341/89882081-994a1c80-dbc6-11ea-8bf0-014a2b29561c.png)

Focused button
![focus-state](https://user-images.githubusercontent.com/13316341/89882077-98b18600-dbc6-11ea-80be-6fcda462232b.png)

**Comment by JAWS-test** [comment]
I do not think so, since the shadow does not fulfill the following:

> Adjacent contrast: The focus indication area has a contrast ratio of at least 3:1 against all adjacent colors for the minimum area or greater, or has a thickness of at least 2 CSS pixels.

And the background changes too little with a contrast difference of 2:1

**Comment by lchandelier** [comment]
Ok, I was not sure especially about the thickness (the 8px thickness about minimum area really confuses me). 

So if I understand well, either I change the grey color to improve contrast with the shadow, or I make the shadow at least 2px thicker? 

I'll change the contrast, I think it's the best solution. But I want to be sure I understand this criterion.

**Comment by JAWS-test** [comment]
The first point is fulfilled, because the focus frame is 1px wide:

> Minimum area: The focus indication area is greater than or equal to a 1 CSS pixel border of the focused control, or has a thickness of at least 8 CSS pixels along the shortest side of the element.

But not the 3rd point.

If I understand the SC correctly, the focus border in your example must be at least 2px wide or the contrast of the background to the unfocused state must be at least 3:1. In the latter case (sufficient contrast) the shadow is not necessary or is not evaluated (contrast and thickness of the shadow do not matter)

**Comment by lchandelier** [comment]
Ok, it's more clear now, thanks for your help!

**Comment by JAWS-test** [comment]
You are also welcome to leave the issue open for others to respond to (possibly with a different opinion). Please note that WCAG 2.2 has not yet been adopted and is still under discussion. Changes may still occur.

**Comment by alastc** [comment]
It is a little hard to tell, because I can't see whether the outline is added around the button (over the background) or inside the button? It might help to see a live version?

The easiest way would be sure is to make the internal background change meet the 3:1 contrast change, possibly reversing the text colour as well.

**Comment by lchandelier** [comment]
I don't have a live version yet. The outline is added around the button.

**Comment by alastc** [comment]
If the dark outline is added around the button, which switches to a slightly lighter colour, I think it would pass because:

- It meets the min area.
- It meets the colour contrast change (white to black).
- It doesn't have adjacent contrast with the button (2:1), but it appears to be 2px.

This issue appears to be a question rather than a request to change the SC, is that right? If this answer is sufficient we can close the issue.

**Comment by lchandelier** [comment]
Yes, that's a question. 

I still have a doubt regarding this point: "The color change for the focus indication area has a contrast ratio of at least 3:1 with the colors of the unfocused state. "

I understand that the background color on focus state (my grey background) must have a contrast ratio of at least 3:1 with the original background color (the black one). Is that correct?

Regarding that all conditions must be true, I don't think this one pass.

**Comment by alastc** [comment]
Hi @lchandelier,

That's why I asked whether the line appeared in / on the button, or around (outside) the button. If it is inside then the change of contrast is not sufficient.

If the black outline appears around/outside the button (on top of the white background, then it would not pass.

See this [codepen for the two examples](https://codepen.io/alastc/pen/JjKbPYY).

**Comment by lchandelier** [comment]
Hi @alastc 
I get it now, thanks a lot for you explainations and codepen.

**Comment by DavidMacDonald** [comment]
why not reverse contrast on the button... on focus

**Comment by mbgower** [comment]
It sounds like it passes the minimum, but depending on the context, if you want to make it obvious where the focus is, you might want to make it a bit more obvious, as @DavidMacDonald suggests :)

**Comment by alastc** [comment]
Hi @lchandelier - the understanding document should be updated with those examples next time it is re-published.
